<launch>
    <arg name="camera_id" default="/camera/color"/>
    <arg name="image_src" default="/image_raw"/>
    
    <arg name="camera_info_src" default="$(find yolov5_d)/config/camera_info.yaml"/>

    <arg name="object_species_file" default="$(find yolov5_d)/data/coco.names"/>
    <arg name="rt_engine_file" default="$(find yolov5_d)/model/yolov5s.engine"/>

    <arg name="score_threshold" default="0.50"/>
    <arg name="nms_threshold" default="0.40"/>

    <node pkg="yolov5_d" name="yolov5_detect" type="yolov5_d" output="screen">
        <param name="base_dir" type="str" value="$(find yolov5_d)"/>
        <param name="image_topic_name" type="str" value="$(arg camera_id)$(arg image_src)"/>
        <param name="camera_info_dir" type="str" value="$(arg camera_info_src)"/>
        <param name="names_file" type="str" value="$(arg object_species_file)"/>
        <param name="engine_file" type="str" value="$(arg rt_engine_file)"/>
    </node>

</launch>
